node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -286,31 +277,40 @@ public void suspend(NodeAgentContext context) {
         }
     }
 
-    private Optional<String> shouldRemoveContainer(NodeAgentContext context, Container existingContainer) {
+    private List<String> shouldRemoveContainer(NodeAgentContext context, Container existingContainer) {
         final NodeState nodeState = context.node().state();
-        if (nodeState == NodeState.dirty || nodeState == NodeState.provisioned) {
-            return Optional.of("Node in state " + nodeState + ", container should no longer be running");
-        }
+        List<String> reasons = new ArrayList<>();
+        if (nodeState == NodeState.dirty || nodeState == NodeState.provisioned)
+            reasons.add("Node in state " + nodeState + ", container should no longer be running");
+
         if (context.node().wantedDockerImage().isPresent() &&
                 !context.node().wantedDockerImage().get().equals(existingContainer.image)) {
-            return Optional.of("The node is supposed to run a new Docker image: "
+            reasons.add("The node is supposed to run a new Docker image: "
                     + existingContainer.image.asString() + " -> " + context.node().wantedDockerImage().get().asString());
         }
-        if (!existingContainer.state.isRunning()) {
-            return Optional.of("Container no longer running");
-        }
+
+        if (!existingContainer.state.isRunning())
+            reasons.add("Container no longer running");
 
         if (currentRebootGeneration < context.node().wantedRebootGeneration()) {
-            return Optional.of(String.format("Container reboot wanted. Current: %d, Wanted: %d",
+            reasons.add(String.format("Container reboot wanted. Current: %d, Wanted: %d",
                     currentRebootGeneration, context.node().wantedRebootGeneration()));
         }
 
-        if (containerState == STARTING) return Optional.of("Container failed to start");
-        return Optional.empty();
+        ContainerResources wantedContainerResources = getContainerResources(context);
+        if (!wantedContainerResources.equalsMemory(existingContainer.resources)) {
+            reasons.add("Container should be running with different memory allocation, wanted: " +
+                    wantedContainerResources.toStringMemory() + ", actual: " + existingContainer.resources.toStringMemory());
+        }
+
+        if (containerState == STARTING)
+            reasons.add("Container failed to start");
+
+        return reasons;
     }
 
-    private void removeContainer(NodeAgentContext context, Container existingContainer, String reason, boolean alreadySuspended) {
-        context.log(logger, "Will remove container: " + reason);
+    private void removeContainer(NodeAgentContext context, Container existingContainer, List<String> reasons, boolean alreadySuspended) {
+        context.log(logger, "Will remove container: " + (reasons.size() == 1 ? reasons.get(0) : "\n" + String.join("\n", reasons)));
```

Comment:
I'd vote for comma-separated rather than newline, as the latter is not so good with grep and is visually extravagant.